master/buildbot/process/buildrequest.py

```diff
@@ -267,7 +267,18 @@ def canBeCollapsed(master, br1, br2):
         if br1['buildsetid'] == br2['buildsetid']:
             defer.returnValue(True)
 
-        # get the buidlsets for each buildrequest
+        # Don't collapse e.g. GitLab merge requests with normal builds
+        bsp1 = yield master.db.buildsets.getBuildsetProperties(br1['buildsetid'])
+        bsp2 = yield master.db.buildsets.getBuildsetProperties(br2['buildsetid'])
```

Comment:
please get the properties for bsp2 only if the bsp1 change has merge_request property (for optimisation)

Reply:
I did that originally, but had trouble avoiding improper collapsing then, as normal builds would then collapse themselves with merge requests :-(